### PR TITLE
Revert pr #5020 (Parquet scan time metrics)

### LINKED ISSUE #5788: Incorrect metrics for ParquetExec
### Describe the bug

For tpch 1G parquet data, we run the query 17 and get the following plan result:
Screen Shot 2023-03-30 at 11 20 05

It takes around 4s to finish the query. However, the metrics for the `ParquetExec` shows it takes more than 7s for total scanning, which is abnormal

### To Reproduce

`cargo run --release --bin tpch -- benchmark datafusion --iterations 3 --path ./benchmarks/data-parquet --format parquet --query 17 --disable-statistics --debug`

It seems this bug is introduced by #5020. The new introduced `time_scanning_total` will count the time elapsed by the parent operators of `ParquetExec`. Then this metrics is meaningless. Therefore, I recommend to remove this metrics.

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by yahoNanJing**
# Which issue does this PR close?

Closes #5788.

# Rationale for this change

For tpch 1G parquet data, we run the query 17 and get the following plan result:
Screen Shot 2023-03-30 at 11 20 05

It takes around 4s to finish the query. However, the metrics for the ParquetExec shows it takes more than 7s for total scanning, which is abnormal. The new introduced time_scanning_total will count the time elapsed by the parent operators of ParquetExec. Then this metrics is meaningless. Therefore, I recommend to remove this metrics.

# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by yahoNanJing** [comment]
Hi @xiaoyong-z and @alamb, could you help have a check?

**Comment by alamb** [comment]
For anyone else following along the original PR that added this change was https://github.com/apache/arrow-datafusion/pull/5020

@yahoNanJing  -- looking at the screen shot you provided, 
228734458-ddd80ca3-f59d-47a6-bd55-141ed31924e3

It looks to me like the parquet file in question is being read using 2 streams (aka the file was opened by two different tasks which are reading it in parallel)

Thus while the wall clock time takes only 4 seconds it may be possible that the total cpu time is actually 7 seconds

You could potentially disable the `repartition_file_scans` option

https://docs.rs/datafusion/latest/datafusion/config/struct.OptimizerOptions.html#structfield.repartition_file_scans

And see if the metrics were more like what you expected

Perhaps @thinkharderdev or @tustvold  have additional thoughts they could share

**Comment by yahoNanJing** [comment]
Thanks @alamb. My previous comment may be misleading. The main controversial is the metrics of **_time_elapsed_scanning_total_** contains the times cost by its parent operators, which may be misleading. By the above metrics, developers may think the bottleneck is the parquet scanning. However, it should be the aggregation whose time is wrongly counted in the metrics **_time_elapsed_scanning_total_**

**Review by thinkharderdev — COMMENTED** [review]
I don't think we should remove this metric. I agree that it can be superficially misleading when the actual bottleneck is an upstream operator, but it's also **possible** for the bottleneck to be parquet scanning. In those cases we would not really have any metric to capture that with this change.

**Comment by alamb** [comment]
>  Thanks @alamb. My previous comment may be misleading. The main controversial is the metrics of time_elapsed_scanning_total contains the times cost by its parent operators, which may be misleading. By the above metrics, developers may think the bottleneck is the parquet scanning. However, it should be the aggregation whose time is wrongly counted in the metrics time_elapsed_scanning_total

I think the parent/child relationships of where time is being spent will be hard to capture in scalar metrics (aka any sort of "elapsed time" metrics). I think to get that level of detail, DataFusion would likely need some sort of tracing / span abstraction that can show the heirarchy

File: datafusion/core/src/physical_plan/file_format/file_stream.rs [hunk 1]
```diff
@@ -232,23 +223,20 @@ impl<F: FileOpener> FileStream<F> {
...
-                                partition_values,
-                            }
-                        }
-                        Ok(None) => return Poll::Ready(None),
-                        Err(e) => {
-                            self.state = FileStreamState::Error;
-                            return Poll::Ready(Some(Err(e)));
+                FileStreamState::Idle => match self.start_next_file().transpose() {
+                    Ok(Some((future, partition_values))) => {
+                        self.file_stream_metrics.time_opening.start();
```
[the remark this answers is not in the document]
Comment (alamb) [reply datafusion/core/src/physical_plan/file_format/file_stream.rs:228]: Filed https://github.com/apache/arrow-datafusion/pull/5898

[the remark this answers is not in the document]
Comment (alamb) [reply datafusion/core/src/physical_plan/file_format/file_stream.rs:228]: https://github.com/apache/arrow-datafusion/pull/5900

**Pull request closed without merging**